app/components/sign/agreement-popup.tsx

```diff
@@ -0,0 +1,53 @@
+import { useCallback } from "react";
+import { useSearchParams } from "@remix-run/react";
+import { AnimatePresence } from "framer-motion";
+import Agreement from "./agreement";
+import { Button } from "../shared";
+
+export const AGREEMENT_POPUP_VISIBLE = "show-agreement-popup";
+
+export default function AgreementPopup({
+  templateName,
+}: {
+  templateName: String;
+}) {
+  const [params, setParams] = useSearchParams();
+  const isMobileView = params.get(AGREEMENT_POPUP_VISIBLE) || false;
+  const handleBackdropClose = useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) => {
+      if (e.target !== e.currentTarget) return;
+      params.delete(AGREEMENT_POPUP_VISIBLE);
+      setParams(params);
+    },
+    [params, setParams]
+  );
+
+  return (
+    <AnimatePresence>
+      {isMobileView ? (
+        <div
```

Comment:
Why not use the Dialog component we have already?

Reply:
I copied this one from the asset page :/

Reply:
@DonKoko can you link the usage of the existing component once?

Reply:
@rajdip-b here is the component: https://github.com/Shelf-nu/shelf.nu/blob/main/app/components/layout/dialog.tsx
If you search for `<Dialog` you should be able to see how we use it, but its pretty straight forward.